master/internal/resourcemanagers/agent/agent.go

```diff
@@ -92,16 +95,31 @@ func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 		socket, ok := msg.Accept(ctx, aproto.MasterMessage{}, true)
 		check.Panic(check.True(ok, "failed to accept websocket connection"))
 		a.socket = socket
+
 		lastColonIndex := strings.LastIndex(msg.Ctx.Request().RemoteAddr, ":")
 		if lastColonIndex == -1 {
 			a.address = msg.Ctx.Request().RemoteAddr
 		} else {
 			a.address = msg.Ctx.Request().RemoteAddr[0:lastColonIndex]
 		}
-		wsm := ws.WriteMessage{Message: aproto.AgentMessage{MasterSetAgentOptions: a.opts}}
+
+		var masterSetAgentOptions aproto.AgentMessage
+		if a.awaitingReconnect && a.agentReattachEnabled {
+			optsCopy := *a.opts
+			optsCopy.ContainersToReattach = a.gatherContainersToReattach(ctx)
+			masterSetAgentOptions = aproto.AgentMessage{MasterSetAgentOptions: &optsCopy}
+		} else {
+			// Quick reconnect after restart/crash.
+			clearAll := map[cproto.ID]aproto.ContainerReattachAck{}
+			a.clearNonReattachedContainers(ctx, clearAll)
+			masterSetAgentOptions = aproto.AgentMessage{MasterSetAgentOptions: a.opts}
+		}
```

Comment:
just a bit clearer that it's the same when it's a first run. also a bit less wasteful.
```suggestion
		var masterSetAgentOptions aproto.AgentMessage
		switch {
		case a.awaitingReconnect && a.agentReattachEnabled:
			optsCopy := *a.opts
			optsCopy.ContainersToReattach = a.gatherContainersToReattach(ctx)
			masterSetAgentOptions.MasterSetAgentOptions = &optsCopy
		case a.awaitingReconnect:
			// Quick reconnect after restart/crash.
			clearAll := map[cproto.ID]aproto.ContainerReattachAck{}
			a.clearNonReattachedContainers(ctx, clearAll)
			masterSetAgentOptions.MasterSetAgentOptions = a.opts
		default:
			masterSetAgentOptions.MasterSetAgentOptions = a.opts
		}
```

Reply:
i explained what i meant by this more clearly further down, so resolving.